Update CoinJoin.md (#1051)
zkSNACKs/WasabiDoc: docs/using-wasabi/CoinJoin.md

---
{
  "title": "CoinJoin",
  "description": "A detailed explanation about how WabiSabi coinjoins work under the hood. This is the Wasabi documentation, an archive of knowledge about the open-source, non-custodial and privacy-focused Bitcoin wallet for desktop."
}
---

# Coinjoin

A [coinjoin](https://en.bitcoin.it/Privacy#CoinJoin) is a special Bitcoin transaction where several peers get together to literally join their coins in a single transaction.
They collaboratively build a transaction where each of them provides some coins as inputs, and fresh addresses as outputs.
The concept has been around since the early days of Bitcoin, and it was formalized by the great Greg Maxwell in [this awesome introductory thread](https://bitcointalk.org/index.php?topic=279249.msg2983902).

The goal is to gain privacy by breaking the link of which input "pays" which output so that none of the outputs cannot be attributed to the owner of the input.

> WabiSabi enables centrally coordinated coinjoins with variable amounts in a trustless (meaning nobody can steal) and private (meaning even the coordinator cannot spy) manner, as described in [WabiSabi Framework report](https://eprint.iacr.org/2021/206).

Reviewer: ```suggestion
WabiSabi enables centrally coordinated coinjoins with variable amounts in a trustless (meaning nobody can steal) and private (meaning even the coordinator cannot spy) manner, as described in [WabiSabi protcol](https://eprint.iacr.org/2021/206).
```

[[toc]]

---

## Coinjoin step-by-step

Launch Wasabi, and open your wallet that contains the coins that you want to coinjoin.
Notice that it is not yet possible to CoinJoin from a hardware wallet, the keys must be "hot" on your computer.

> By default, Wasabi starts automatically coinjoining the funds whenever there's more than 0.01 BTC in total. 

Reviewer: ```suggestion
By default, Wasabi starts automatically coinjoining the funds whenever there's more than 0.01 BTC non private in total.
```

> Wallets with less than 0.01 BTC are in _PlebStop-mode_, where funds are not coinjoined automatically.

Reviewer: ```suggestion
Wallets with less than or equal to 0.01 BTC are in _PlebStop_ mode, where funds are not coinjoined automatically.
```

> User can change the limit from the settings and boost the wallets privacy for one session if wanted.

Reviewer: This is not in the settings yet, and why only for one session?

> In the first coinjoin, a 0.3% coordinator fee will be taken from UTXO's bigger than 0.01 BTC.

Reviewer: ```suggestion
In the first coinjoin, a 0.3% coordinator fee will be taken from UTXOs bigger than 0.01 BTC.
```

Smaller ones don't pay coordinator fees at all, according to _PlebsDontPay_ rule.

> The round starts either as soon as 100 inputs have been registered, or after one hour has elapsed since the last round.

Reviewer: Is it limited to 100 inputs or participants?

Author: 100 inputs. The coordinator doesn't know about participants.
https://tumblebit.slack.com/archives/CFBJX6MN0/p1644575203714149

Just leave Wasabi running in the background of your computer.

Remixing is free, as well as coinjoining coins 1 hop from coinjoin, although, Bitcoin network fees still do apply.

> So if you send a payment and receive a change output, Wasabi will automatically remix it without you having to pay coordinator fees again. 

Reviewer: Saying that a change output will be automatically remixed is wrong. There are many variables for a coin to be selected for coinjoin.

> The recipient of the payment will also be able to coinjoin the funds he recieved with only the cost of network fees, as long he/she is also using Wasabi or other WabiSabi compatible Bitcoin wallet.

Reviewer: > as long he/she is also using Wasabi

This is not correct, I might be using another bip 39 compatible wallet, and recover it in Wasabi, so I can coinjoin without paying coordinator fee.

Reviewer 2: ```suggestion
The recipient of the payment will not have to pay coordinator fee, as long he is making coinjoins with the same coordinator.
```

Reviewer 2: ```suggestion
The recipient of the payment will not have to pay coordinator fee, as long he is making coinjoins with the same coordinator.
```

> Once a coin achieves enough ambiguity, the corresponding amount will show up in the software's main view as "private coins".

Reviewer: ```suggestion
Once a coin achieves enough anonymity, the corresponding amount will show up in the software's main view as "private coins".
```

The concept has been around since the early days of Bitcoin, and it was formalized by the great Greg Maxwell in [this awesome introductory thread](https://bitcointalk.org/index.php?topic=279249.msg2983902).

The goal is to gain privacy by breaking the link of which input "pays" which output so that none of the outputs can be attributed to the owner of the input.
WabiSabi enables centrally coordinated coinjoins with variable amounts in a trustless (meaning nobody can steal) and private (meaning even the coordinator cannot spy) manner, as described in [WabiSabi paper](https://eprint.iacr.org/2021/206).

[[toc]]

---

## Coinjoin step-by-step

Launch Wasabi, and open your wallet that contains the coins that you want to coinjoin.
Notice that it is not yet possible to coinjoin from a hardware wallet, the keys must be "hot" on your computer.

By default, Wasabi starts automatically coinjoining the funds whenever there's more than 0.01 BTC non private in total.
Wallets with less than or equal to 0.01 BTC are in _PlebStop_ mode, where funds are not coinjoined automatically.

> User can change the limit from settings or turn coinjoin on manually by pressing the play button in main views "music box".

Reviewer: ```suggestion
User can change the limit from the settings or turn coinjoin on manually by pressing the play button of the music box in the main view.
```

> A 0.3% coordinator fee will be taken from fresh UTXOs bigger than 0.01 BTC.

Reviewer: ```suggestion
A 0.3% coordinator fee will be taken from fresh coins bigger than 0.01 BTC.
```

Smaller ones don't pay coordinator fees at all, according to _PlebsDontPay_ rule.

> The round starts either as soon as 300 inputs have been registered, or after the input registration time has ran out and as long as there's at least 150 inputs registered.

Reviewer 2: ```suggestion
The round starts either as soon as enough inputs have been registered, or after the input registration time is reached and as long as the minimum number of inputs are registered.
```

Just leave Wasabi running in the background of your computer, as coinjoining takes time.

Remixing is free, as well as coinjoining coins 1 hop from coinjoin, although, Bitcoin network fees still do apply.
So if you send a payment and receive a change output, Wasabi will automatically remix it without you having to pay coordinator fees again. 
The recipient of the payment will also be able to coinjoin the funds he recieved with only the cost of network fees, as long he/she is also using Wasabi or other WabiSabi compatible Bitcoin wallet.

Once a coin achieves enough privacy, the corresponding amount will show up in the software's main view as "private coins".

## WabiSabi protocol step-by-step

> WabiSWabi protocol requires 5 steps to successfully create and broadcast a coinjoin transaction to the bitcoin network.

Reviewer 2: ```suggestion
WabiSabi protocol requires 5 steps to successfully create and broadcast a coinjoin transaction to the bitcoin network.
```

### Input registration

During the [input registration](/FAQ/FAQ-UseWasabi.md#what-is-happening-in-the-input-registration-phase), automatic coin selection robot selects which coins will be registered for coinjoin.

> These coins need to be confirmed on the Bitcoin blockchain unless they are from a Wasabi coinjoin and the robot is re-register them.

Reviewer: ```suggestion
These coins need to be confirmed on the Bitcoin blockchain unless they are from a Wasabi coinjoin and the robot is re-registering them.
```

> These coins need to be confirmed on the Bitcoin blockchain unless they are from a Wasabi coinjoin and the robot is re-register them.

Reviewer 2: ```suggestion
These coins need to be confirmed on the Bitcoin blockchain.
```

> In the background, Wasabi generates an input proofs, signatures over a challenge message with the private key that locks up the coins.

Reviewer: ```suggestion
In the background, Wasabi generates input proofs, signatures over a challenge message with the private key that locks up the coins.
```

Reviewer 2: ```suggestion
In the background, Wasabi generates input proofs, a signatures over a challenge message with the private key that locks up the coins.
```

## Coinjoin step-by-step

Launch Wasabi, and open your wallet that contains the coins that you want to coinjoin.
Notice that it is not yet possible to coinjoin from a hardware wallet, the keys must be "hot" on your computer.

By default, Wasabi starts automatically coinjoining the funds whenever there's more than 0.01 BTC non private in total.
Wallets with less than or equal to 0.01 BTC are in _PlebStop_ mode, where funds are not coinjoined automatically.
User can change the limit from settings or turn coinjoin on manually by pressing the play button in main views "music box".

A 0.3% coordinator fee will be taken from fresh UTXOs bigger than 0.01 BTC.
Smaller ones don't pay coordinator fees at all, according to _PlebsDontPay_ rule.

> The round starts either as soon as 300 inputs have been registered, or after the input registration time has ran out and as long as there's at least 150 inputs registered.

Reviewer: I am not sure if we should mention the exact numbers here, because they might change.
